### PR TITLE
Refactor versioning

### PR DESCRIPTION
**Pull request by shimastripe**
After #112 , merge this PR.
## WHAT
- Use the structure for versioning

### CODE REVIEW

**Comment by dtan4** [comment]
plz rebase this branch to the latest master.

**Comment by shimastripe** [comment]
Rebase this branch to the latest master! Plz check.

File: _templates/skeleton/version/version.go.tmpl [hunk 1]
```diff
@@ -8,43 +8,60 @@ import (
...
+	version := &Version{}
+
+	if err := version.initialize(c); err != nil {
+		return nil, err
+	}
+
+	return version, nil
+}
+
+func (self *Version) initialize(c *gin.Context) error {
```
Comment (dtan4) [inline _templates/skeleton/version/version.go.tmpl]: This method should extract out of `Version` structure, or include in `NewVersion` method.

**Review by dtan4 — CHANGES_REQUESTED** [review]

File: _templates/skeleton/version/version.go.tmpl (hunk 1 above)
Comment (shimastripe) [inline _templates/skeleton/version/version.go.tmpl]: OK, I fixed.

File: _templates/skeleton/version/version.go.tmpl [hunk 2]
```diff
@@ -8,43 +8,51 @@ import (
...
+	if _, err := strconv.Atoi(strings.Join(strings.Split(targetVersion, "."), "")); err != nil {
+		return nil, err
 	}
 
-	return ver, nil
+	return &Version{targetVersion}, nil
 }
 
-func Range(left string, op string, right string) bool {
+func (self Version) Range(op string, right string) bool {
```
Comment (dtan4) [inline _templates/skeleton/version/version.go.tmpl]: Change receiver type as pointer.

`func (v *Version) Range(op string, right string) bool`

**Review by dtan4 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**